### PR TITLE
CMR-8788 - Update Ingest and Search Documentation for Generics

### PR DESCRIPTION
**Pull request by jceaser**
* Adding a new variable to publish all Generic Document Version numbers as a list
* Clean up of all the Generic Search and Ingest Documents

### CODE REVIEW

File: ingest-app/docs/api.md [hunk 1]
```diff
@@ -815,7 +823,39 @@ curl -XPUT \
   -H "Authorization: Bearer XXXX" \
   %CMR-ENDPOINT%/providers/PROV1/services/service123 \
   -d \
-"{\"Name\": \"GESDISC_HL2SS\", \"LongName\": \"Harmony Level 2 Subsetting Service (HL2SS) for GES DISC\", \"Version\": \"1.3.1\", \"Type\": \"Harmony\", \"Description\": \"Endpoint for subsetting L2 Subsetter via Harmony\", \"URL\": { \"Description\": \"PROJECT HOME PAGE\", \"URLValue\": \"https:\/\/harmony.earthdata.nasa.gov\" }, \"ServiceKeywords\": [ { \"ServiceCategory\": \"EARTH SCIENCE SERVICES\", \"ServiceTopic\": \"DATA MANAGEMENT\/DATA HANDLING\", \"ServiceTerm\": \"SUBSETTING\/SUPERSETTING\" } ], \"ServiceOrganizations\": [ { \"Roles\": [ \"ORIGINATOR\" ], \"ShortName\": \"NASA\/JPL\/PODAAC\", \"LongName\": \"Physical Oceanography Distributed Active Archive Center, Jet Propulsion Laboratory, NASA\" } ], \"MetadataSpecification\": { \"URL\": \"https:\/\/cdn.earthdata.nasa.gov\/umm\/service\/v1.5.0\", \"Name\": \"UMM-S\", \"Version\": \"1.5.0\"}}"
+"{
+  \"Name\": \"GESDISC_HL2SS\",
```
Comment (eudoroolivares2016) [inline ingest-app/docs/api.md:827]: This is far more readable

**Review by eudoroolivares2016 — APPROVED** [review]

File: common-app-lib/src/cmr/common_app/static.clj [hunk 2]
```diff
@@ -245,16 +245,21 @@
                 cmr-example-collection-id (str "C1234567-" site-example-provider)
                 doc-type (nth (re-find #"/(.*)/" (str page)) 1)
                 generic-doc-body (read-generic-markdown doc-type)
-                generic-doc-toc (read-generic-markdown-toc (gdocs/all-generic-docs-toc doc-type options))]
+                generic-doc-toc (-> doc-type
+                                    (gdocs/all-generic-docs-toc options)
+                                    (read-generic-markdown-toc)
```
Comment (ygliuvt) [inline common-app-lib/src/cmr/common_app/static.clj:250]: no need to have parenthesis here and the next line.

**Review by ygliuvt — APPROVED** [review]

File: common-app-lib/src/cmr/common_app/static.clj (hunk 2 above)
Comment (jceaser) [reply common-app-lib/src/cmr/common_app/static.clj:250]: Is that our style requirement? It reminds me to much of ruby where function () are optional. I like my functions to look like functions :)
Having said that, I am making the change now.

**Review by mmorahan — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

File: common-lib/src/cmr/common/generics_documentation.clj [hunk 3]
```diff
@@ -114,3 +114,14 @@
   (-> generic-doc-toc
       (string/replace #"\s*<\/li>\n?<\/ul>$" "")
       (string/replace-first #"<ul>\s*\n?\s*<li>" "")))
+
+(defn generic-document-versions->markdown
+  "Creates a markdown list of all the Generic Documents with the lattest version
```
Comment (zimzoom) [inline common-lib/src/cmr/common/generics_documentation.clj:119]: typo, lattest -> latest

File: common-lib/src/cmr/common/config.clj [hunk 4]
```diff
@@ -219,13 +219,7 @@
   "This is the feature toggle for the new document pipeline prototype, as well as serving as
    the base truth list of approved document types. For tests to process, grid is required,
```
Comment (zimzoom) [inline common-lib/src/cmr/common/config.clj:220]: I believe some tests use data quality summary, too

Comment (jceaser) [reply common-lib/src/cmr/common/config.clj:220]: added a test

**Review by zimzoom — APPROVED** [review]

**Pull request merged**